### PR TITLE
Add WeakRef::new() for creating a new, empty WeakRef

### PR DESCRIPTION
**Pull request by sdroege**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
:+1:

File: src/object.rs [hunk 1]
```diff
@@ -963,6 +963,20 @@ pub struct WeakRef<T: ?Sized> {
     marker: PhantomData<*const T>,
 }
 
+impl<T> WeakRef<T> {
+    pub fn new() -> Self {
+        unsafe {
+            let weak = WeakRef {
+                inner: Arc::new(WeakRefInner(mem::zeroed())),
+                thread: Some(thread::current().id()),
```
Comment (tmiasko) [inline src/object.rs]: None? It should be safe regardless of thread.

File: src/object.rs [hunk 2]
```diff
@@ -963,6 +963,20 @@ pub struct WeakRef<T: ?Sized> {
     marker: PhantomData<*const T>,
 }
 
+impl<T> WeakRef<T> {
+    pub fn new() -> Self {
```
Comment (tmiasko) [inline src/object.rs]: Maybe Default::default instead / in addition?

File: src/object.rs (hunk 1 above)
Comment (sdroege) [reply src/object.rs]: True

File: src/object.rs (hunk 2 above)
Comment (sdroege) [reply src/object.rs]: Either would be fine. What would people prefer?

**Comment by sdroege** [comment]
Updated

**Comment by sdroege** [comment]
@GuillaumeGomez should be good to go :)

**Comment by GuillaumeGomez** [comment]
Indeed. Thanks!

**Pull request merged**